Neos.Neos/Classes/Fusion/Cache/ContentCacheFlusher.php

```diff
@@ -92,23 +92,41 @@ class ContentCacheFlusher
      * model or the Neos Publishing Service.
      *
      * @param NodeInterface $node The node which has changed in some way
+     * @param Workspace $targetWorkspace An optional workspace to flush
      * @return void
      * @throws \Neos\ContentRepository\Exception\NodeTypeNotFoundException
      */
-    public function registerNodeChange(NodeInterface $node)
+    public function registerNodeChange(NodeInterface $node, Workspace $targetWorkspace = null)
     {
         $this->tagsToFlush[ContentCache::TAG_EVERYTHING] = 'which were tagged with "Everything".';
 
         if (empty($this->workspacesToFlush[$node->getWorkspace()->getName()])) {
             $this->resolveWorkspaceChain($node->getWorkspace());
         }
 
+        if ($targetWorkspace !== null && empty($this->workspacesToFlush[$targetWorkspace->getName()])) {
+            $this->resolveWorkspaceChain($targetWorkspace);
+        }
+
         if (!array_key_exists($node->getWorkspace()->getName(), $this->workspacesToFlush)) {
             return;
         }
 
+        $this->registerAllTagsToFlushForNodeInWorkspace($node, $node->getWorkspace());
+        if ($targetWorkspace !== null) {
+            $this->registerAllTagsToFlushForNodeInWorkspace($node, $targetWorkspace);
+        }
+    }
+
+    /**
+     * @param NodeInterface $node
+     * @param Workspace $workspace
+     */
+    protected function registerAllTagsToFlushForNodeInWorkspace(NodeInterface $node, Workspace $workspace)
```

Comment:
Nitpick: Would be nice to have complete return types for newly introduced methods - so void in this case

Reply:
```suggestion
    protected function registerAllTagsToFlushForNodeInWorkspace(NodeInterface $node, Workspace $workspace): void
```